### ISSUE TITLE
inconsistency in filename case sensitivity handling

### ISSUE THREAD

**Issue by gperciva** [Tarsnap/tarsnap#167]
When one refers to a filename or directory, the case sensitivity depends on the filesystem.  So on OSX (with the default filesystem),

    keyfile /users/me/tarsnap.key

works just fine.  However, the `include` and `exclude` commands are always case sensitive, since it uses `fnmatch()` in the C layer, not the filesystem layer.

I'll update the docs, but I'm also leaning towards printing a warning if `keyfile` finds a file due to case-insensitivity when the string doesn't match.

**Comment by marrie** [comment]
I like that idea, however is it possible to let tarsnap be the same case sensitivity as the osx? For example on mac os you might have hfs journaled, or the same but case sensitive. Rather then case insensitive. I don't know how this would work, however it is a thought I've ben considering, since I've ben helping the author of this posttry and find this bug.

**Comment by gperciva** [comment]
It gets complicated -- OSX can be case-sensitive if you configure it that way.  Also, I'm not aware of a safe and reliable way to check if a certain filesystem is case-sensitive or not.  Also, it could occur that somebody's `/` directory is case-insensitive, but `/Users/` is case-sensitive, but `/Users/me` is case-insensitive... I'll need to look into various possibilities.

**Comment by marrie** [comment]
Oh yeah I kind of forgot about that, oops?  If the "warning" were to take place I guess the user would have to change the "exclude" function in their script. Correct? Just some thoughts here.
>

**Comment by marrie** [comment]
Hello there.  I'm noticing that when I delete an archive  it takes for ever, actually it doesn't happen. I'm noticing the following prompt.


load: 2.66  cmd: tarsnap 68371 waiting 4.57u 0.28s
load: 2.60  cmd: tarsnap 68371 waiting 4.68u 0.32s

Now, I should add I restored a CCC partition today.  Can this have something why the archive appeared to freeze  when deleting?  I don't want to reregister my machine as I don't want to lose all I uploaded.  I made sure my key was there. All of my archives list and a back up did fine. So what did I do this time?

Thanks.

**Comment by gperciva** [comment]
Hi, I've sent you an email so that we can discuss the "delete an archive" issue separately.  Let's keep this github issue for discussing filename case sensitivity.

**Comment by marrie** [comment]
Good idea.  I think maybe the message should be what comes up if you are in debug mode, something like this.

"tarsnap, unable to exclude /aath/to/file. Reason: cases do not match. Please check with your system administrator for more info." or at least something like that.

The message could of corse be changed but at least it's a step in the right direction.
